pkg/operator/operator.go

```diff
@@ -591,12 +607,7 @@ func (o *Operator) sync(ctx context.Context, key string) error {
 	config.SetTelemetryMatches(o.telemetryMatches)
 	config.SetRemoteWrite(o.remoteWrite)
 
-	var proxyConfig manifests.ProxyReader
-	proxyConfig, err = o.loadProxyConfig(ctx)
-	if err != nil {
-		klog.Warningf("using proxy config from CMO configmap: %v", err)
-		proxyConfig = config
-	}
+	var proxyConfig = getProxyReader(func() (*ProxyConfig, error) { return o.loadProxyConfig(ctx) }, config)
```

Comment:
Just a thought 
```suggestion
	var proxyConfig = getProxyReader(ctx, o.loadProxyConfig, config)
```
The code can read "simpler" if context is passed to `getProxyReader` which then passes the `ctx` to `proxySupplier` that accepts a `context`.